src/ray/core_worker/task_interface.cc

```diff
@@ -25,17 +26,64 @@ ActorHandle::ActorHandle(
   inner_.set_is_direct_call(is_direct_call);
 }
 
-ActorHandle::ActorHandle(const ActorHandle &other)
-    : inner_(other.inner_), new_actor_handles_(other.new_actor_handles_) {}
+ActorHandle::ActorHandle(ActorHandle &parent, bool in_band) {
+  std::unique_lock<std::mutex> guard(parent.mutex_);
+  inner_ = parent.inner_;
+  class ActorHandleID new_actor_handle_id;
+  if (in_band) {
```

Comment:
the naming of `in_band` here seems not quite clear. 

How about renaming it to:

`Fork():`  corresponds to the case where `in_band` is `true`
`ForkForSerialization():` corresponds to the case where `in_band` is `false`

The other constructor below can be renamed to: 
`Deserialize():` creates an actor handle from a serialized handle that is generated by `ForkForSerialization` above.

Reply:
@raulchen what do you think?

Reply:
I changed this to instead have a `Fork()` and `ForForSerialization` that each return a `unique_ptr`. Please take a look.